core/bin/snapshot_creator/src/main.rs

```diff
@@ -0,0 +1,115 @@
+use anyhow::Context as _;
+
+use zksync_dal::connection::DbVariant;
+use zksync_dal::ConnectionPool;
+use zksync_object_store::{ObjectStore, ObjectStoreFactory};
+use zksync_types::snapshots::{SnapshotChunk, SnapshotFactoryDependency, SnapshotStorageKey};
+use zksync_utils::ceil_div;
+
+async fn run(blob_store: Box<dyn ObjectStore>, pool: ConnectionPool) {
```

Comment:
It makes sense propagating errors here using `anyhow::Result` + `anyhow::Context` to add contextual info.